File: week11/2/Prob4.py
from typing import List

class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if not strs:
            return ""
        minStr = min(strs, key = len)
        for i, x in enumerate(minStr):
            for other in strs:
                if other[i] != x:
                    return minStr[:i]
        return minStr

File: week11/2/Prob4.cpp
#include<iostream>
#include<vector>

using namespace std;

class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string lcp = "";
        // sort strs
        sort(strs.begin(), strs.end()); 
        
        string first_str = strs[0];
        string last_str = strs[strs.size() - 1];

        for (int i=0; i< first_str.size(); i++) {
            if (first_str[i] == last_str[i])
                lcp += first_str[i];
            else
                break;
        }

        return lcp;
    }
};